packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart

```diff
@@ -1010,47 +1005,58 @@ class ProfileReportEntry {
 
 /// Profiling information for a range of token positions in a [Script].
 class ProfileReportRange {
-  ProfileReportRange._fromJson(Script script, Map<String, dynamic> json)
-      : metadata = ProfileReportMetaData._(
-          sampleCount: json[_kProfileKey][_kMetadataKey][_kSampleCountKey],
-        ),
-        inclusiveTicks = json[_kProfileKey][_kInclusiveTicksKey].cast<int>(),
-        exclusiveTicks = json[_kProfileKey][_kExclusiveTicksKey].cast<int>(),
-        lines = json[_kProfileKey][_kPositionsKey]
-            .map<int>(
-              // It's possible to get a synthetic token position which will
-              // either be a negative value or a String (e.g., 'ParallelMove'
-              // or 'NoSource'). We'll just use -1 as a placeholder since we
-              // won't display anything for these tokens anyway.
-              (e) => e is int
-                  ? script.getLineNumberFromTokenPos(e) ?? _kNoSourcePosition
-                  : _kNoSourcePosition,
-            )
-            .toList() {
+  ProfileReportRange._fromJson(Script script, _ProfileReportRangeJson json) {
+    final sampleCount = json.metadata[_kSampleCountKey] as int;
```

Comment:
should sampleCount be a getter in the extension type?

Reply:
done